src/Microsoft.AspNetCore.Mvc.Testing/WebHostBuilderExtensions.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.Testing.Internal;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.Testing
+{
+    /// <summary>
+    /// Testing extensions for creating an configuring an <see cref="IWebHostBuilder"/> for end to end in memory tests.
+    /// </summary>
+    public static class WebHostBuilderTestingExtensions
+    {
+        private static readonly string CreateDefaultBuilder = nameof(CreateDefaultBuilder);
+
+        /// <summary>
+        /// Initializes a new <see cref="IWebHostBuilder"/> instance by calling a method 
+        /// <c>public static CreateDefaultBuilder(string [] args)</c> on the
+        /// entry point class for the assembly. Tipically, <c>Program</c>.
+        /// </summary>
+        /// <typeparam name="TStartup">The type of the Startup class.</typeparam>
+        /// <param name="args">The arguments passed to CreateDefaultBuilder, typically from <c>Program.Main(string [] args)</c>.</param>
+        /// <returns>An <see cref="IWebHostBuilder"/> instance.</returns>
+        public static IWebHostBuilder FromStartup<TStartup>(string[] args)
+        {
+            var factory = typeof(TStartup).Assembly.EntryPoint.DeclaringType.GetMethod(CreateDefaultBuilder, BindingFlags.Static | BindingFlags.Public);
+            if (factory == null)
+            {
+                return null;
```

Comment:
Should this do a `Console\Debug.WriteLine` saying no `IWebHostBuilder` was discovered and a default one would be used? Assuming of course the test runner \ VS etc prints the message.

Reply:
I don't think those things get printed today and on VS it will be hard to discover anyways.

Reply:
I'm going to change this to throw instead (more obvious)